python/ray/autoscaler/v2/instance_manager/subscribers/instance_launcher.py

```diff
@@ -1,160 +0,0 @@
-import logging
-import math
-from collections import defaultdict
-from concurrent.futures import ThreadPoolExecutor
-from typing import List
-
-from ray.autoscaler._private.constants import (
-    AUTOSCALER_MAX_CONCURRENT_LAUNCHES,
-    AUTOSCALER_MAX_LAUNCH_BATCH,
-)
-from ray.autoscaler.v2.instance_manager.instance_storage import (
-    InstanceStorage,
-    InstanceUpdatedSubscriber,
-)
-from ray.autoscaler.v2.instance_manager.node_provider import NodeProvider
-from ray.core.generated.instance_manager_pb2 import Instance, InstanceUpdateEvent
-
-logger = logging.getLogger(__name__)
-
-
-class InstanceLauncher(InstanceUpdatedSubscriber):
-    """InstanceLauncher is responsible for provisioning new instances."""
-
-    def __init__(
-        self,
-        instance_storage: InstanceStorage,
-        node_provider: NodeProvider,
-        max_concurrent_requests: int = math.ceil(
-            AUTOSCALER_MAX_CONCURRENT_LAUNCHES / float(AUTOSCALER_MAX_LAUNCH_BATCH)
-        ),
-        max_instances_per_request: int = AUTOSCALER_MAX_LAUNCH_BATCH,
-    ) -> None:
-        self._instance_storage = instance_storage
-        self._node_provider = node_provider
-        self._max_concurrent_requests = max_concurrent_requests
-        self._max_instances_per_request = max_instances_per_request
-        self._executor = ThreadPoolExecutor(max_workers=1)
-        self._launch_instance_executor = ThreadPoolExecutor(
-            max_workers=self._max_concurrent_requests
-        )
-
-    def notify(self, events: List[InstanceUpdateEvent]) -> None:
-        # TODO: we should do reconciliation based on events.
-        has_new_request = any(
-            [event.new_instance_status == Instance.UNKNOWN for event in events]
-        )
-        if has_new_request:
-            self._executor.submit(self._may_launch_new_instances)
-
-    def _may_launch_new_instances(self):
-        new_instances, _ = self._instance_storage.get_instances(
-            status_filter={Instance.UNKNOWN}
-        )
-
-        if not new_instances:
-            logger.debug("No instances to launch")
-            return
-
-        queued_instances = []
```

Comment:
the logic here for converting QUEUED to REQUESTED has been now moved to this PR https://github.com/ray-project/ray/pull/42973/files